include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -449,6 +528,204 @@ namespace picongpu
                     }
                 }
 
+                void writeOpenPMDFile(uint32_t currentStep)
+                {
+                    auto buffer = std::make_unique<HostBuffer<float_X, DIM3>>(DataSpace<DIM3>(
+                        parameters::nTheta,
+                        parameters::nPhi,
+                        transitionRadiation::frequencies::nOmega));
+                    auto dataBox = buffer->getDataBox();
+
+                    for(unsigned int index_direction = 0; index_direction < transitionRadiation::parameters::nObserver;
+                        ++index_direction)
+                    {
+                        // theta
+                        const int i = index_direction / parameters::nPhi;
+                        // phi
+                        const int j = index_direction % parameters::nPhi;
+
+                        for(unsigned int k = 0; k < transitionRadiation::frequencies::nOmega; ++k)
+                        {
+                            dataBox({static_cast<int>(i), static_cast<int>(j), static_cast<int>(k)})
+                                = static_cast<float_X>(
+                                    theTransRad[index_direction * transitionRadiation::frequencies::nOmega + k]);
+                        }
+                    }
+
+                    std::stringstream filename;
+                    filename << filenamePrefix << "_%T." << fileExtension;
+
+                    ::openPMD::Series series(filename.str(), ::openPMD::Access::CREATE);
+
+                    ::openPMD::Extent extent
+                        = {static_cast<unsigned long int>(transitionRadiation::frequencies::nOmega),
+                           static_cast<unsigned long int>(parameters::nPhi),
+                           static_cast<unsigned long int>(parameters::nTheta)};
+                    ::openPMD::Offset offset = {0, 0, 0};
+                    ::openPMD::Datatype datatype = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset dataset{datatype, extent};
+
+                    auto mesh = series.iterations[currentStep].meshes["transitionradiation"];
```

Comment:
```suggestion
                    auto iteration = series.writeIterations()[currentStep];
                    auto mesh = iteration.meshes["transitionradiation"];
```

This API has better support for ADIOS2 steps.